File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;
class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        
        ArrayList<String> c = new ArrayList<>();
        if(cacheSize == 0){
            return cities.length * 5;
        }
        
        for(int i = 0; i < cities.length; i++){
            String city = cities[i].toUpperCase();
            
            if(c.contains(city)){
                c.remove(city);
                c.add(city);
                answer+=1;
            }
            else{
            
                if(c.size() == cacheSize){
                    c.remove(0);
            }
                c.add(city);
                answer+=5;
        }
            
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque

def solution(cacheSize, cities):
    answer = 0
    
    memory = deque()
    cities = [city.lower() for city in cities]

    for i in cities:
        if i in memory:
            answer+=1
            memory.remove(i)
            memory.append(i)
        else:
            answer+=5
            memory.append(i)
            if len(memory) > cacheSize:
                memory.popleft()
    return answer